cli/flox/src/commands/init/node.rs

```diff
@@ -285,20 +297,66 @@ impl Node {
     }
 
     /// Try to find nodejs (alone) that satisfies constraints in package.json
+    #[instrument(skip_all, fields(progress = "searching for compatible node versions"))]
     async fn try_find_compatible_nodejs(
         flox: &Flox,
         versions: &PackageJSONVersionsUnresolved,
     ) -> Result<Option<PackageJSONVersion>> {
         let PackageJSONVersionsUnresolved { node, .. } = versions;
 
+        debug!(
+            version = node.as_ref().unwrap_or(&"unspecified".to_string()),
+            "trying to find compatible node version"
+        );
+        // Prioritize newer major versions by putting them in order. The bare "nodejs"
+        // packages goes last because an old nodejs version that matches against it will
+        // pin the package group to an old page.
+        let nodejs_packages = ["nodejs_23", "nodejs_22", "nodejs_20", "nodejs_18", "nodejs"];
         let found_node = match node {
             Some(node_version) => {
-                match try_find_compatible_package(flox, "nodejs", Some(node_version)).await? {
-                    None => Some(PackageJSONVersion::Unavailable),
-                    Some(result) => Some(PackageJSONVersion::Found(result)),
+                let resolution_futs = nodejs_packages.iter().map(|attr_path| {
+                    try_find_compatible_package(flox, attr_path, Some(node_version))
+                });
+                // The result of `join_all` is a `Vec` of the result of resolving each
+                // future in the order that they were provided e.g. the order of
+                // `nodejs_packages`. This means the results are returned in order of
+                // preference for latest package version *when using the live catalog*.
+                // However, since the order in which the futures are resolved at runtime
+                // isn't deterministic, the order in which the resolutions are recorded
+                // in the test JSON file is also not deterministic. That means you may get
+                // results returned in the wrong order during tests. To mitigate that
+                // we just put the results in a map and pull them back out in the
+                // correct order.
+                let resolutions = join_all(resolution_futs)
+                    .await
+                    .into_iter()
+                    .filter_map(|res| res.unwrap_or_default())
```

Comment:
nit - I was confused about the default value causes filtering out, and this would have made it more clear to me
```suggestion
                    .filter_map(|res| res.unwrap_or(None))
```

Reply:
The suggestion will get a clippy violation but I can leave a comment

Reply:
Afaict it does not trigger a clippy violation

Reply:
Don't know what to tell you, that's why I did it in the first place

Reply:
Weird
```
> git diff --no-ext-diff
diff --git a/cli/flox/src/commands/init/node.rs b/cli/flox/src/commands/init/node.rs
index 929780e18..1dacff968 100644
--- a/cli/flox/src/commands/init/node.rs
+++ b/cli/flox/src/commands/init/node.rs
@@ -330,7 +330,7 @@ impl Node {
                 let resolutions = join_all(resolution_futs)
                     .await
                     .into_iter()
-                    .filter_map(|res| res.unwrap_or_default())
+                    .filter_map(|res| res.unwrap_or(None))
                     .map(|pkg| (pkg.attr_path.to_string(), pkg))
                     .collect::<BTreeMap<String, _>>();
                 let resolutions_sorted = {
@@ -368,7 +368,7 @@ impl Node {
     /// This will perform a search to determine if a requested version is
     /// available.
     async fn get_nvmrc_version(flox: &Flox, path: &Path) -> Result<Option<NVMRCVersion>> {
-        let nvmrc = path.join(".nvmrc");
+        let nvmrc = path.join(&".nvmrc");
         if !nvmrc.exists() {
             return Ok(None);
         }
> pushd cli; cargo clippy; popd
~/flox/main/cli ~/flox/main
warning: the borrowed expression implements the required traits
   --> flox/src/commands/init/node.rs:371:31
    |
371 |         let nvmrc = path.join(&".nvmrc");
    |                               ^^^^^^^^^ help: change this to: `".nvmrc"`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrows_for_generic_args
    = note: `#[warn(clippy::needless_borrows_for_generic_args)]` on by default

warning: `flox` (bin "flox") generated 1 warning (run `cargo clippy --fix --bin "flox"` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.53s
```